ICSharpCode.Decompiler/CSharp/Resolver/CSharpConversions.cs

```diff
@@ -1229,6 +1229,41 @@ List<OperatorInfo> GetApplicableConversionOperators(ResolveResult fromResult, IT
 		}
 		#endregion
 
+		#region Implicit Span Conversion
+
+		public bool IsImplicitSpanConversion(IType fromType, IType toType)
+		{
+			// An implicit span conversion permits array_types, System.Span<T>, System.ReadOnlySpan<T>,
+			// and string to be converted between each other 
+			// see https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-14.0/first-class-span-types#span-conversions
+
+			switch (fromType)
+			{
+				case ArrayType { Dimensions: 1, ElementType: var elementType }:
+					if (toType.IsKnownType(KnownTypeCode.SpanOfT) ||
+						toType.IsKnownType(KnownTypeCode.ReadOnlySpanOfT))
+					{
+						return ImplicitReferenceConversion(elementType, toType.TypeArguments[0], 0)
+							|| IdentityConversion(elementType, toType.TypeArguments[0]);
```

Comment:
IdentityConversion is the cheaper check, so we usually try that one first.
Also you don't need the `, 0` (which just made me look up "what's that extra parameter")